ksql-serde/src/main/java/io/confluent/ksql/serde/delimited/KsqlDelimitedDeserializer.java

```diff
@@ -46,17 +48,17 @@
       Type.STRING, s -> s
   );
 
-  private final Schema schema;
+  private final ConnectSchema schema;
   private final ProcessingLogger recordLogger;
 
   KsqlDelimitedDeserializer(
-      final Schema schema,
+      final PersistenceSchema schema,
       final ProcessingLogger recordLogger
   ) {
-    this.schema = Objects.requireNonNull(schema, "schema").schema();
+    this.schema = Objects.requireNonNull(schema, "schema").getConnectSchema();
```

Comment:
it's a little weird to me that `PersistenceSchema#getConnectSchema` returns `ConnectSchema` instead of either `Schema` or `KsqlSchema`. In general, I see quite a few places around the code that handle `ConnectSchema` instead of just `Schema` and I feel like that's error prone.

Reply:
It's deliberate, and I explained why in the main PR.  The `Schema` interface has both mutable and immutable implementations and any schema instance can be a mixture of them both.

I don't like mutable types, so the explicit use of the type is there as a way of documenting that we're after an immutable schema.

Once the code is all in and I've switched `KsqlSchema` so it doesn't use a connect `Schema` instance internally, then the only place the connect schema will be used is in the serde classes. At this point I'll remove all the `ConnectSchema` stuff most likely.